base/floatfuncs.jl

```diff
@@ -105,24 +105,30 @@ end
 
 function _signif_og(x, digits, base)
     if base == 10
-        oftype(x, 10. ^ floor(log10(abs(x)) - digits + 1.))
+        e = floor(log10(abs(x)) - digits + 1.)
+        og = oftype(x, exp10(abs(e)))
     elseif base == 2
-        oftype(x, 2. ^ floor(log2(abs(x)) - digits + 1.))
+        e = exponent(abs(x)) - digits + 1.
+        og = oftype(x, exp2(abs(e)))
```

Comment:
that should be `ldexp(1.0,abs(e))`: however I think this currently requires `e` to be an integer.